lib/_http_client.js

```diff
@@ -840,7 +842,7 @@ function _deferToConnect(method, arguments_, cb) {
 
   const callSocketMethod = () => {
     if (method)
-      this.socket[method].apply(this.socket, arguments_);
+      FunctionPrototypeApply(this.socket[method], this.socket, arguments_);
```

Comment:
```suggestion
      ReflectApply(this.socket[method], this.socket, arguments_);
```

Reply:
Good catch, `ReflectApply` is a better `FunctionPrototypeApply` :+1: